### DISCUSSION TITLE
AT28C16 supported?

### DISCUSSION THREAD

**Discussion by n9jcv** [TomNisbet/TommyPROM#43] [Q&A] [answered]
So I am at the point with the 8 bit that I need to program a AT28C16.  Can the tommyprom do it?

What changes might I need to make?  If this has already been documed can you please provide a link?

Thank you very much!

**Comment by TomNisbet** [comment, marked as answer]
I haven't tried it, but it should work with the standard 28C code.  The chip does not support block writes, so change the constructor in TommyPROM.ino to read:
`PromDevice28C  prom(2 * 1024L, 0, 10, true);`
The second parameter specifies a block size of zero, which will use the code to default to byte writes instead of blocks.

For the hardware, the chip is a 24 pin package instead of a 28 pin.  If you wire the data, control, and address lines as shown in the schematic it will be fine.  You are just missing some upper address lines that will be left unconnected on the shift registers.

Let me know if you run into any difficulties.
Tom

**Reply by n9jcv** [reply]
Tom, 
Thanks for the quick reply.  I will give this a try and write back about the results so others can benefit.  I appreciate your help.  It may be a couple of weeks until I am able to do this.

**Reply by n9jcv** [reply]
Just to confirm, and for others, the above does work with AT18C16.  Thank you 😊

**Reply by TomNisbet** [reply]
Glad it worked for you!
